src/index.ts

```diff
@@ -336,6 +337,44 @@ function suppressNoSubscriptionExistsWarning<T>(f: () => T): T {
   }
 }
 
+/**
+ * Another VERY hacky monkeypatch to make sure that we can take extra parameters to certain alchemy subscriptions
+ * I hate doing this, but the other option is to fork web3-core and I think for now this is better. Any web3-core
+ * update to this method will have to be manually duplicated by us here.
+ */
+const web3CoreSubscriptions = require("web3-core-subscriptions"); // tslint:disable-line:no-var-requires
+const web3CoreErrors = require("web3-core-helpers").errors; // tslint:disable-line:no-var-requires
+web3CoreSubscriptions.subscription.prototype._validateArgs = function (
```

Comment:
We might consider: instead of throwing out the old prototype method, wrap it:
```ts
const subscriptionPrototype = web3CoreSubscriptions.subscription.prototype;
const oldValidateArgs = subscriptionPrototype._validateArgs;

subscriptionPrototype._validateArgs = function (args: any) {
  // ...
  if (![ /* ... */ ].includes(this.subscriptionMethod) {
    oldValidateArgs.call(this, args);
  }
}
```
This has a bit of an advantage that it doesn't need to copy the error logic and so is a little bit more resilient to changes in the original (it also means we don't need to import `web3-core-helpers`).

Reply:
Yeah I agree with the advantage here. In this particular case I also edited the error handling to include the names of our alchemy methods. Previously if an alchemy method ran into this error it would show "undefined" as the method.

Reply:
Ohh you're right, I did miss that difference. Hm sure, I guess that makes sense, you can keep it as you think best. It's pretty hard to spot that that's the one line that's edited from the original, so maybe leave a comment on it.

Reply:
Switched to wrapping and removed the error handling. It's not relevant to this PR anyway.